lib/file-source/tests/glob.rs

```diff
@@ -0,0 +1,70 @@
+use file_source::paths_provider::{glob::Glob, PathsProvider};
+use std::collections::HashSet;
+
+#[test]
+fn test_glob_include_plain() -> Result<(), Box<dyn std::error::Error>> {
+    let include_patterns = ["tests/files/foo.log".to_owned()];
+    let exclude_patterns = [];
+    let glob = Glob::new(&include_patterns, &exclude_patterns)?;
+
+    let paths = glob.paths()?;
+
+    assert_eq!(
+        paths.into_iter().collect::<HashSet<_>>(),
+        ["./tests/files/foo.log"]
+            .iter()
+            .map(std::path::PathBuf::from)
+            .collect::<HashSet<_>>()
+    );
```

Comment:
I'd tend to do this the other way:
```suggestion
    paths.sort();
    assert_eq!(paths, ["./tests/files/foo.log"]);
```
(sorting is not strictly needed for this test, but would be for the others)